File: Java_Version/src/Tile.java
import java.util.List;
import java.util.ArrayList;

enum sea_type { sea, ship, hit, miss};

public class Tile {

	
	private	int coord_x;
	private int coord_y;
	private sea_type type;
	
	public Tile() {
		
	};
	
	public Tile( int coo_x, int coo_y, sea_type typ) {
		this.coord_x = coo_x;
		this.coord_y = coo_y;
		this.type = type;
	};
	
	public List<Integer> getCoord(){
		List<Integer> cords = new ArrayList<Integer>();
	    cords.add(this.coord_x);
	    cords.add(this.coord_y);
	 
	    return(cords);
	};
	
	public void setCoord( int xx, int yy) {
		this.coord_x = xx;
		this.coord_y = yy;
	};
	
	public sea_type getType() {
		return this.type;
	};
	
	public void setType(sea_type type) {
		this.type = type;
	};
	
	public void draw() {
		
		if( this.type == sea_type.sea) {
				System.out.print("~");
				//break;
		}
		else if( this.type == sea_type.ship) {
				System.out.print("s");
				//break;
		}
		else if( this.type == sea_type.hit) {
				System.out.print("X");
				//break;
		}
		else if( this.type == sea_type.miss) {
				System.out.print("o");
				//break;
		}
	}
	
}

File: Cpp_Version/tile.cpp
#include "stdafx.h"
#include "tile.h"
#include <iostream>

Tile::Tile() {

}

Tile::Tile(int coo_x, int coo_y, sea_type type) {
	coord_x = coo_x;
	coord_y = coo_y;
	this->type = type;
}

std::vector<int> Tile::getCoord() {
	std::vector<int> coords(2);
	coords[0] = this->coord_x;
	coords[1] = this->coord_y;

	return coords;
}

void Tile::setCoord(int cx, int cy) {
	this->coord_x = cx;
	this->coord_y = cy;
}

sea_type Tile::getType() {
	return this->type;
}

void Tile::setType(sea_type type) {
	this->type = type;
}

void Tile::draw() {

	switch (type) {
		case sea:
			std::cout << "~";
			break;
		case ship:
			std::cout << "s";
			break;
		case hit:
			std::cout << "X";
			break;
		case miss:
			std::cout << "o";
			break;
	}
}

